spikesorters/yass/yass.py

```diff
@@ -0,0 +1,280 @@
+        
+        
+        #################################################################
+        #################### SAVE RAW INT16 data ########################
+        #################################################################
+        # ALESSIO Look at Kilosort 
+        # There is alrady an extractor se.Mea1kRecordingExtractor()
+        # all the functions are there already to concatenate in time;
+        # multi-recording time extractor;
+        
+        # save binary file; THIS IS FROM KILOSORT
+        input_file_path = os.path.join(output_folder, 'data.bin')
+        
+        recording.write_to_binary_dat_format(input_file_path, 
+                                             dtype='int16', 
+                                             chunk_mb=500) # time_axis=0,1 for C/F order
+        
+        
+        #################################################################
+        #################### SAVE UPDATED CONFIG FILE ###################
+        #################################################################
+        fname_config = os.path.join(output_folder,
+                                   'config.yaml')
+        with open(fname_config, 'w') as file:
+            documents = yaml.dump(self.params, file)
+        
+        #self.fname_config = fname_config
+        
+        # ALESSIO:
+        # Expose more config file parameters that are sensiitive:
+        #  e.g. spike width; smallest cluster; min firing rates;
+        #  
+            
+    # FUNCTION TO RUN YASS 
+    #def _run(self,recording, output_folder):  # SOMETIMES want to access more information from recording in
+                                               # this step
+    def _run(self, recording, output_folder):
+        '''
+        '''
+        recording = recover_recording(recording)  # allows this to run on multiple jobs (not just multi-core)
+        
+        if 'win' in sys.platform and sys.platform != 'darwin':
+            shell_cmd = '''
+                        yass sort {config}
+                    '''.format(config=os.path.join(output_folder,'config.yaml'))
+        else:
+            shell_cmd = '''
+                        #!/bin/bash
+                        yass sort {config}
+                    '''.format(config=os.path.join(output_folder,'config.yaml'))
+
+        shell_script = ShellScript(shell_cmd, 
+                                   script_path=os.path.join(output_folder,self.sorter_name),
+                                   log_path=os.path.join(output_folder,self.sorter_name+'.log'), 
+                                   verbose=self.verbose)
+        shell_script.start()
+
+        retcode = shell_script.wait()
+
+        if retcode != 0:
+            raise Exception('yass returned a non-zero exit code')
+    
+    # Alessio might not want to put here; 
+    # better option to have a parameter "tune_nn" which 
+    def train(self, recording, output_folder):
```

Comment:
Could you document a bit when to use this "train" function.

In spikesorter, the ed user is suposed to do a simple `sorting = run_yass(recording)`.
Which internaly create a yassSroter instance and run it.
Does the "train" occur before or inside the "run" ?

Reply:
Yes, we can add it there, let's discuss the best option.

Reply:
Added now.  Default behaviour retrains NNs directly. 

Option to provide previously trained NNs:

ss.run_yass(rec, '/home/cat/Downloads/test2', neural_nets_path='/home/cat/tmp/nn_train')